sdk/invoke.js

```diff
@@ -33,24 +34,24 @@ const invoke = async function invoke(accountId, funcName, apiKey, funcData) {
     logger.debug('Invoke response', { statusCode, headers, body });
     return res;
   } catch (err) {
-    let parsedError;
-    try {
-      parsedError = JSON.parse(err.error);
-    } catch (nestedErr) {
-      logger.debug('Failed to parse JSON response', { nestedErr });
-    }
-
-    if (parsedError) {
-      const errorOrMsg = parsedError.error || parsedError.message;
-      if (parsedError.stack && errorOrMsg) {
-        throw new Error(`${errorOrMsg}\n${parsedError.stack}`);
+    const fallbackError = err.error || err;
+    if (err instanceof StatusCodeError) {
+      const requestId = err.response.headers['x-binaris-request-id'];
+      let parsedError;
+      try {
+        parsedError = JSON.parse(err.error);
+      } catch (parseError) {
```

Comment:
doesn't `JSON.parse(err.error)` only make sense if `err.error` exists?
if err.error is missing, there will always be a parse error and then parsedError.error will be the fallbackError which is really err which is a very convoluted way to write err :-/

I think the entire section can be way more readable.
##
```javascript
...
  } catch (err) {
    const errorMessage = [];
    const requestId = err.response.headers['x-binaris-request-id'];
    if (requestId) {
      errorMessage.push = `RequestId: ${requestId}`;
    }
    if (err instanceof StatusCodeError) { // maybe not needed
      if (err.error) {
        try {
          const subError = JSON.parse(err.error);
          errorMessage.push(subError.error || subError.message);
          if (subError.stack) {
            errorMessage.push(subError.stack);
          }
        } catch (_) {
           errorMessage.push(err.error);
        }
      } else {
        errorMessage.push(err.message);
      }
      throw new Error(errorMessage.join('\n'));
    }
    throw err;
  }
```
That said, I would refactor this entirely, to use APIError which is really ErrorWithRequestId,
and name the function handling this accordingly (something about handle nested error)

Reply:
Note that `StatusCodeError.error` is supposedly deprecated (response.body instead?): https://github.com/request/promise-core/blob/master/lib/errors.js - it contains the response body - it might fail if it's `undefined` or if it's non-JSON (such as legacy Java responses).